crates/optimism/payload/src/builder.rs

```diff
@@ -677,3 +757,67 @@ where
         Ok(None)
     }
 }
+
+/// Extracts commitment from access list entries, pointing to 0x420..022 and validates them
+/// against supervisor.
+///
+/// If commitment present pre-interop tx rejected.
+///
+/// returns (is_valid, is_recoverable)
+/// is_recoverable to set to false if we can't recover from error, in case supervisor is down,
+/// connection is broken or transaction is invalid deterministically.
+pub fn is_cross_tx_valid<Primitives: OpPayloadPrimitives>(
+    tx: &Primitives::SignedTx,
+    client: &SupervisorValidator,
+    safety_level: SafetyLevel,
+    timestamp: u64,
+) -> (bool, bool) {
+    if tx.access_list().is_none() {
+        return (true, true);
+    }
+    let access_list = tx.access_list().unwrap();
+    let inbox_entries =
+        SupervisorValidator::parse_access_list(access_list).cloned().collect::<Vec<_>>();
+    if !inbox_entries.is_empty() {
+        match tokio::task::block_in_place(move || {
```

Comment:
we definitely don't want to block in place for an rpc call. this is ok for now though, think best way to go about this is adding a new trait method `PayloadBuilder::try_build_non_blocking` or call somewhere else or smthg - but is out of scope of this pr

Reply:
We could pass the executor inside, but this will leak executor abstraction all around, so I didn't like it